File: MyDent.Services/Database/Appointment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using MyDent.Model.Enums;

namespace MyDent.Services.Database
{
    public class Appointment
    {
        [Key]
        public int Id { get; set; }

        public int PatientId { get; set; }

        [ForeignKey("PatientId")]
        public User Patient { get; set; } = null!;

        public int DoctorId { get; set; }

        [ForeignKey("DoctorId")]
        public Doctor Doctor { get; set; } = null!;

        public int DentalServiceId { get; set; }

        [ForeignKey("DentalServiceId")]
        public DentalService DentalService { get; set; } = null!;

        [Required]
        public DateTime ScheduledAt { get; set; }

        // Snapshot from DentalService at booking time — price/duration must not
        // silently change if the service's price is edited later.
        [Required]
        public int DurationMinutes { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }

        [Required]
        public AppointmentStatus Status { get; set; } = AppointmentStatus.Pending;

        [MaxLength(500)]
        public string? CancellationReason { get; set; }

        public int? CancelledByUserId { get; set; }

        [ForeignKey("CancelledByUserId")]
        public User? CancelledByUser { get; set; }

        public DateTime? CancelledAt { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<AppointmentStatusHistory> StatusHistory { get; set; } = new List<AppointmentStatusHistory>();
        public Review? Review { get; set; }
        public Payment? Payment { get; set; }
        public ICollection<Notification> Notifications { get; set; } = new List<Notification>();
    }
}

File: UI/mydent_desktop/lib/models/appointment.dart
import 'package:json_annotation/json_annotation.dart';

part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  final int? id;
  final int? patientId;
  final String? patientName;
  final int? doctorId;
  final String? doctorName;
  final int? dentalServiceId;
  final String? dentalServiceName;
  final DateTime? scheduledAt;
  final int? durationMinutes;
  final double? price;

  // Raw int matching AppointmentStatus (see enums.dart) — no JsonStringEnumConverter on the API.
  final int? status;
  final String? cancellationReason;
  final int? cancelledByUserId;
  final DateTime? cancelledAt;
  final DateTime? createdAt;

  Appointment({
    this.id,
    this.patientId,
    this.patientName,
    this.doctorId,
    this.doctorName,
    this.dentalServiceId,
    this.dentalServiceName,
    this.scheduledAt,
    this.durationMinutes,
    this.price,
    this.status,
    this.cancellationReason,
    this.cancelledByUserId,
    this.cancelledAt,
    this.createdAt,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) =>
      _$AppointmentFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}
